modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -90,6 +100,109 @@ public void addValueType(DiagnosticPos pos, String typeName) {
         }
     }
 
+    public void addArrayType(DiagnosticPos pos, int dimensions) {
+        BLangType eType;
+        if (!this.typeNodeListStack.empty()) {
+            List<TypeNode> typeNodeList = this.typeNodeListStack.peek();
+            eType = (BLangType) typeNodeList.get(typeNodeList.size() - 1);
+            typeNodeList.remove(typeNodeList.size() -1);
+        } else {
+            eType = (BLangType) this.typeNodeStack.pop();
+        }
+        BLangArrayType arrayTypeNode = (BLangArrayType) TreeBuilder.createArrayTypeNode();
+        arrayTypeNode.pos = pos;
+        arrayTypeNode.etype = eType;
+        arrayTypeNode.dimensions = dimensions;
+
+        if (!this.typeNodeListStack.empty()) {
```

Comment:
Shall we please factor these lines out to a private method, such as "addType(TypeNode)". This is being used by many methods, we should update those places as well.

Reply:
Will do.